src/Http/ResponseEmitter.php

```diff
@@ -0,0 +1,274 @@
+ * @license   https://github.com/zendframework/zend-diactoros/blob/master/LICENSE.md New BSD License
+ */
+namespace Cake\Http;
+
+use Cake\Core\Configure;
+use Cake\Log\Log;
+use Psr\Http\Message\ResponseInterface;
+use Zend\Diactoros\RelativeStream;
+use Zend\Diactoros\Response\EmitterInterface;
+
+/**
+ * Emits a Response to the PHP Server API.
+ *
+ * This emitter offers a few changes from the emitters offered by
+ * diactoros:
+ *
+ * - It logs headers sent using CakePHP's logging tools.
+ * - Cookies are emitted using setcookie() to not conflict with ext/session
+ */
+class ResponseEmitter implements EmitterInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function emit(ResponseInterface $response, $maxBufferLength = 8192)
+    {
+        $file = $line = null;
+        if (headers_sent($file, $line)) {
+            $message = "Unable to emit headers. Headers sent in file=$file line=$line";
+            if (Configure::read('debug')) {
+                trigger_error($message, E_USER_WARNING);
+            } else {
+                Log::warn($message);
+            }
+        }
+
+        $this->emitStatusLine($response);
+        $this->emitHeaders($response);
+        $this->flush();
+
+        $range = $this->parseContentRange($response->getHeaderLine('Content-Range'));
+
+        if (is_array($range)) {
+            $this->emitBodyRange($range, $response, $maxBufferLength);
+
+            return;
+        }
+
+        $this->emitBody($response, $maxBufferLength);
+
+        if (function_exists('fastcgi_finish_request')) {
+            fastcgi_finish_request();
+        }
+    }
+
+    /**
+     * Emit the message body.
+     *
+     * @param \Psr\Http\Message\ResponseInterface $response The response to emit
+     * @param int $maxBufferLength The chunk size to emit
+     * @return void
+     */
+    protected function emitBody(ResponseInterface $response, $maxBufferLength)
+    {
+        $body = $response->getBody();
+
+        if (!$body->isSeekable()) {
+            echo $body;
+
+            return;
+        }
+
+        $body->rewind();
+        while (!$body->eof()) {
+            echo $body->read($maxBufferLength);
+        }
+    }
+
+    /**
+     * Emit a range of the message body.
+     *
+     * @param array $range The range data to emit
+     * @param \Psr\Http\Message\ResponseInterface $response The response to emit
+     * @param int $maxBufferLength The chunk size to emit
+     * @return void
+     */
+    protected function emitBodyRange(array $range, ResponseInterface $response, $maxBufferLength)
+    {
+        list($unit, $first, $last, $length) = $range;
+
+        $body = $response->getBody();
+
+        if (!$body->isSeekable()) {
+            $contents = $body->getContents();
+            echo substr($contents, $first, $last - $first + 1);
+
+            return;
+        }
+
+        $body = new RelativeStream($body, $first);
```

Comment:
Will there be no problems if this class couples itself to `Zend\Diactoros` in future?